### DISCUSSION TITLE
[ZK Stack] - zk-environment container doesn't start

### DISCUSSION THREAD

**Discussion by ghost** [zkSync-Community-Hub/zksync-developers#679] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I have started the ZK Stack on my server using the zk_inception CLI, but it generated docker-compose.yml file with the zk container that has strange env.vars. This container doesn't start on my server:
...  
zk:
    image: "ghcr.io/matter-labs/zk-environment:latest2.0-lightweight"
    security_opt:
      - seccomp:unconfined
    command: tail -f /dev/null
    volumes:
      - .:/usr/src/zksync
      - /usr/src/cache:/usr/src/cache
      - /var/run/docker.sock:/var/run/docker.sock
      - ./hardhat-nodejs:/root/.cache/hardhat-nodejs
    environment:
      - CACHE_DIR=/usr/src/cache
      - SCCACHE_CACHE_SIZE=50g
      - SCCACHE_GCS_BUCKET=matterlabs-infra-sccache-storage
      - SCCACHE_GCS_SERVICE_ACCOUNT=gha-ci-runners@matterlabs-infra.iam.gserviceaccount.com
      - SCCACHE_ERROR_LOG=/tmp/sccache_log.txt
      - SCCACHE_GCS_RW_MODE=READ_WRITE
      - CI=1
      - GITHUB_WORKSPACE=$GITHUB_WORKSPACE
    env_file:
      - ./.env
    extra_hosts:
      - "host:host-gateway"
    profiles:
      - runner
    network_mode: host

### Expected Behavior

1. zk container working as expected.
2. Clean up unused env.vars

### Current Behavior

Here's my generated docker-compose.yml:
https://pastebin.com/KQD14JH5

It was generated with the following command:
```
        zk_inception ecosystem create \
            -v --ecosystem-name genlayer \
            --chain-id 71271 \
            --l1-batch-commit-data-generator-mode rollup \
            --start-containers true \
            --link-to-code /home/runner/work/zk/zk/ \
            --chain-name genlayer-blockchain \
            --wallet-creation localhost \
            --prover-mode no-proofs \
            --l1-network localhost \
            --base-token-address 0x0000000000000000000000000000000000000001 \
            --base-token-price-nominator 1 \
            --base-token-price-denominator 1
```

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Reply by bxpana** [reply]
Will take a look into this and see if I can replicate.

**Comment by ghost** [comment]
Any updates on this?

**Reply by bxpana** [reply]
Still working on this. Ran into some issues with the setup, but should be able to get back to you this week

**Reply by bxpana** [reply]
Ok got it working. So then you're running `zk_inception ecosystem init --dev` and you're getting an error? Or is the main concern the unused env.vars? 

You're also using a custom origin of the `zksync-era` repo right? Have you changed anything there?

**Comment by ghost** [comment]
Yes, I'm running `zk_inception ecosystem init --dev` command, but the ZK container doesn't start on my server. The environment variables in the generated docker-compose.yml includes some strange or unused environment variables that might be causing the problem. Only reth and PostgreSQL database containers are started.
I've just cloned zksync-era repo in my home directory and didn't change anything.

**Reply by bxpana** [reply]
The team is looking into removing it from the generated docker-compose files, but it shouldn't cause any issues with your container with it being there. 

When you say it doesn't start, what happens? I would try maybe just using the option for cloning from the repo instead of your clone in your home directory. If that doesn't work, then I would delete the files and start fresh with again the zksync-repo from the source and not your clone.

**Comment by ghost** [comment]
When you say it doesn't start, what happens?
```
        cd /tmp
        zk_inception ecosystem create \
            -v --ecosystem-name genlayer \
            --chain-id 71271 \
            --l1-batch-commit-data-generator-mode rollup \
            --start-containers true \
            --link-to-code /home/runner/work/zk/zk/ \
            --chain-name genlayer-blockchain \
            --wallet-creation localhost \
            --prover-mode no-proofs \
            --l1-network localhost \
            --base-token-address 0x0000000000000000000000000000000000000001 \
            --base-token-price-nominator 1 \
            --base-token-price-denominator 1
```
I run this command, and it generates the following docker-compose file:
https://pastebin.com/bF1fUJw5
But the zk-environment:latest2.0-lightweight container doesn't start...
Then I init ZK Stack, it works fine:
`zk_inception ecosystem init --dev --verbose`
Logs: https://pastebin.com/5BAaCV8W
And start the ZK Stack server:
`cd /tmp/genlayer && zk_inception server`
Logs: https://pastebin.com/VjvqAPR3

**Reply by bxpana** [reply]
What if you try to run it according to this tutorial? Does it work then? https://code.zksync.io/tutorials/custom-zk-chain

It's a bit different startup and will help us see if it's the instructions or something else.
